restapi.go

```diff
@@ -1221,17 +1221,19 @@ func (s *Session) Channel(channelID string) (st *Channel, err error) {
 // ChannelEdit edits the given channel
 // channelID  : The ID of a Channel
 // name       : The new name to assign the channel.
-func (s *Session) ChannelEdit(channelID, name string) (st *Channel, err error) {
-
-	data := struct {
-		Name string `json:"name"`
-	}{name}
+func (s *Session) ChannelEdit(channelID, name string) (*Channel, error) {
+	return s.ChannelEditComplex(&ChannelEdit{
+		Name: name,
+		ID:   channelID,
+	})
+}
 
-	body, err := s.RequestWithBucketID("PATCH", EndpointChannel(channelID), data, EndpointChannel(channelID))
+// ChannelEditComplex edits an existing channel, replacing the parameters entirely with ChannelEdit struct
+func (s *Session) ChannelEditComplex(c *ChannelEdit) (st *Channel, err error) {
```

Comment:
Sorry to be a pain, could we change this method signature to:

ChannelEditComplex(channel ID string, c *ChannelEdit)

We can then remove the ID field from the struct.

This would bring this API closer to the ChannelMessageSendComplex method.

Thanks!

Reply:
Not a pain, thank you for the feedback. This is a learning experience for me.

The current signature is mimicking ChannelMessageEditComplex:
`func (s *Session) ChannelMessageEditComplex(m *MessageEdit) (st *Message, err error)`

and the MessageEdit struct:
```
type MessageEdit struct {
	Content *string       `json:"content,omitempty"`
	Embed   *MessageEmbed `json:"embed,omitempty"`

	ID      string
	Channel string
}
```

Should ChannelMessageEditComplex be rewritten to be inline with your suggestion?
Adding channelID seems to match the other functions.

Reply:
I see that changing ChannelMessageEditComplex will break the existing API.